### ISSUE TITLE
nginx_conf - Wildcard include-support not working

### ISSUE THREAD

**Issue by rndmh3ro** [inspec/inspec#2485]
### Description

The `include` in the `nginx.conf` directive breaks then inspec nginx_conf resource.

### InSpec and Platform Version
Inspec 1.51.0

Tested on CentOS7 and Arch Linux

### Replication Case

1. create a minimal nginx.conf:
```
http {
    client_body_buffer_size 1k;
    include /etc/nginx/conf.d/*.conf;
    }
```

2. create a .conf file in `/etc/nginx/conf.d/` with the following contents:
```
client_header_buffer_size 1k;
```

3. run inspec shell. Empty response.
```
inspec> nginx_conf.params
=> {}
```
4. comment the include-line and run inspec shell again. Still not the expected result but better than before.
```
inspec> nginx_conf.params
=> {"http"=>[{"client_body_buffer_size"=>[["1k"]]}]}
```

Bonus:

If I change the include line to any of theses:
```
    include "/etc/nginx/conf.d/*.conf";
    include '/etc/nginx/conf.d/*.conf';
    include conf.d/*.conf;
    include "conf.d/*.conf";
    include 'conf.d/*.conf';
```
I get these messages:
```
find_files(): exit 1 from `sh -c 'find /etc/nginx/"/etc/nginx/conf.d/*.conf"'`
find_files(): exit 1 from `sh -c 'find /etc/nginx/'/etc/nginx/conf.d/*.conf''`
find_files(): exit 1 from `sh -c 'find /etc/nginx/"conf.d/*.conf"'`
```

Aha! Link: https://chef.aha.io/features/SH-2540

**Comment by rndmh3ro** [comment]
CCing @arlimus and @jerryaldrichiii.

**Comment by rndmh3ro** [comment]
Same problem as https://github.com/chef/inspec/issues/2278 and https://github.com/chef/inspec/issues/1998 and maybe https://github.com/chef/inspec/issues/2157

**Comment by adamleff** [comment]
Yup, looks like a bug. I'll log it as such.

**Comment by aaronlippold** [comment]
I am seeing this issue as well but am surprised we are seeing it again - I thought we had squashed this already

**Comment by jerryaldrichiii** [comment]
@aaronlippold @rndmh3ro 

I was unable to reproduce this. Could you look at the below and verify it fails for you:

Here is the `Vagrantfile` I used to spin up the NGINX server:

```ruby
minimal_nginx_conf = <<-EOF
worker_processes  1;

events {
    worker_connections  1024;
}

http {
    client_body_buffer_size 1k;
    include /etc/nginx/conf.d/*.conf;
}
EOF

extra_nginx_conf = <<-EOF
client_header_buffer_size 1k;
EOF

Vagrant.configure("2") do |config|
  config.vm.box = "bento/centos-7"
  config.vm.network "private_network", ip: "192.168.33.10"

  config.vm.provision "shell", inline: <<-SHELL
    yum install -y epel-release
    yum install -y nginx vim
    systemctl enable nginx
  SHELL

  config.vm.provision "shell", inline: <<-SHELL
    echo '#{minimal_nginx_conf}' > /etc/nginx/nginx.conf
    echo '#{extra_nginx_conf}' > /etc/nginx/conf.d/inspec_testing.conf
  SHELL

  config.vm.provision "shell", inline: <<-SHELL
    systemctl start nginx
  SHELL
end
```

Here is the output from `inspec shell -t ssh://vagrant:vagrant@192.168.33.10`:

```
[~/Projects/github.com/chef/inspec]$ git checkout v1.51.0
HEAD is now at 1d2ecdb... Bump version to 1.51.0 by Expeditor
[~/Projects/github.com/chef/inspec]$ bundle exec inspec -v
1.51.0

Your version of InSpec is out of date! The latest version is 2.0.17.
[~/Projects/github.com/chef/inspec]$ bundle exec inspec shell -t ssh://vagrant:vagrant@192.168.33.10
Welcome to the interactive InSpec Shell
To find out how to use it, type: help

You are currently running on:

    OS platform: centos
    OS family: redhat
    OS release: 7.4.1708

inspec> nginx_conf.params
=> {"worker_processes"=>[["1"]],
 "events"=>[{"worker_connections"=>[["1024"]]}],
 "http"=>
  [{"client_body_buffer_size"=>[["1k"]], "client_header_buffer_size"=>[["1k"]]}]}
inspec>
```

**Comment by jerryaldrichiii** [comment]
I was able to verify that quotes surrounding the path does not work however.

See: https://github.com/chef/inspec/pull/2726

**Comment by jerryaldrichiii** [comment]
I think I know the issue here:

```
[root@localhost vagrant]# sh -c 'find /etc/nginx/conf.d/*.conf'
find: ‘/etc/nginx/conf.d/*.conf’: No such file or directory
[root@localhost vagrant]# touch /etc/nginx/conf.d/foobar.conf
[root@localhost vagrant]# sh -c 'find /etc/nginx/conf.d/*.conf'
/etc/nginx/conf.d/foobar.conf
```

TL;DR: If there are no files found with `/etc/nginx/conf.d/*.conf` then the command exits 1.

**Comment by jerryaldrichiii** [comment]
Closing this due to it's age and the resolution in https://github.com/inspec/inspec/pull/2726 

Please reopen if this is still an issue.

**Comment by jerryaldrichiii** [comment]
Closing this due to it's age and the resolution in #2726

Please reopen if this is still an issue.

**Comment by aaronlippold** [comment]
This seems to be coming up again in current releases of inspec

**Comment by james-stocks** [comment]
@aaronlippold - closed issues are not reviewed by Engineering. If the issue in the current InSpec is exactly the same reproduction case as the description for this issue then we should re-open this issue.
If it's just somewhat similar to this issue then please open a new issue.

**Comment by aaronlippold** [comment]
Not sure if it's exact but I think the issue is a windows issue that wasn't
tested from the orig fix. Not sure if the solution will be the same between
platforms.
